src/main/java/edu/agh/wfiis/designpatterns/patternsUsageExample/DesignPattern.java

```diff
@@ -1,102 +1,117 @@
-    }
-
-    public String getExample() {
-        return example;
-    }
-
-    public void setExample(String example) {
-        this.example = example;
-    }
-
-    public boolean isCreational() {
-        return creational;
-    }
-
-    public void setCreational(boolean creational) {
-        this.creational = creational;
-    }
-
-    public boolean isStructural() {
-        return structural;
-    }
-
-    public void setStructural(boolean structural) {
-        this.structural = structural;
-    }
-
-    public boolean isBehavioral() {
-        return behavioral;
-    }
-
-    public void setBehavioral(boolean behavioral) {
-        this.behavioral = behavioral;
-    }
-
-    public String getPatternCode() {
-        return patternCode;
-    }
+    public String refactor(String badCode){
+        /* some magic involving patternCode happens here, irrelevant from this example perspective...*/
 
-    public void setPatternCode(String patternCode) {
-        this.patternCode = patternCode;
+        String goodCode = "this code is very good";
+        return goodCode;
     }
 
-    public DesignPattern(String id, String name, String description, String example, String patternCode, boolean creational, boolean structural, boolean behavioral) {
-        this.id = id;
-        this.name = name;
-        this.description = description;
-        this.example = example;
+    private DesignPattern(Details patternDetails, String patternCode) {
+        this.patternDetails = patternDetails;
         this.patternCode = patternCode;
-        this.creational = creational;
-        this.structural = structural;
-        this.behavioral = behavioral;
     }
 
-    public String refactor(String badCode){
-        /* some magic involving patternCode happens here, irrelevant from this example perspective...*/
-
-        String goodCode = "this code is very good";
-        return goodCode;
+    @Component
+    public static class Builder implements CreatingDesignPattern, HavingName {
+        @Override
+        public DesignPattern create() {
+            return new DesignPattern(
+                    new Details("1", name(), "builder pattern", "some builder example", Type.CREATIONAL),
+                    "some builder code"
+            );
+        }
+
+        @Override
+        public String name() {
+            return "builder";
+        }
+    }
+
+    @Component
+    public static class Decorator implements CreatingDesignPattern, HavingName {
+        @Override
+        public DesignPattern create() {
+            return new DesignPattern(
+                    new Details("2", name(), "decorator pattern", "some decorator example", Type.CREATIONAL),
+                    "some decorator code"
+            );
+        }
+
+        @Override
+        public String name() {
+            return "decorator";
+        }
+    }
+
+    @Component
+    public static class Strategy implements CreatingDesignPattern, HavingName {
```

Comment:
Obsolete class. Same functionality could be achieved by declaring method like:

`public static DesignPattern createStrategy() {`
`   return new DesignPattern(`
`                    new Details("3", name(), "strategy pattern", "some strategy example", Type.BEHAVIORAL),`
`                    "some strategy code"`
`            );`
`}`

Or better without obsolete class Details (which encapsulate nothing in the end and its not Builder patter either)

Reply:
Totally agree on that. There is no need to create new classes with changing member's values if you don't change behavior of these classes.